src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Logging/AbstractRazorLoggerFactory.AggregateLogger.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Razor.Logging;
+
+internal abstract partial class AbstractRazorLoggerFactory
+{
+    private class AggregateLogger(ImmutableArray<ILogger> loggers) : ILogger
+    {
+        private ImmutableArray<ILogger> _loggers = loggers;
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            foreach (var logger in _loggers)
+            {
+                if (logger.IsEnabled(logLevel))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public void Log<TState>(LogLevel logLevel, TState state, Exception? exception, Func<TState, Exception?, string> formatter)
+        {
+            foreach (var logger in _loggers)
+            {
+                if (logger.IsEnabled(logLevel))
```

Comment:
I'm assuming this was always the case, but some of our loggers also check `IsEnabled` before logging. Should we remove that check there?

Reply:
Will do something about this in a follow up PR. With an interpolated string handler, hopefully the Log call can be avoided entirely in the right circumstances